### ISSUE TITLE
NaN val loss

### ISSUE THREAD

**Issue by FredHaa** [The-Learning-Machines/Audiomer-PyTorch#4]
Hi again

After a reinstall with Ubuntu 20.04 and CUDA 11.3 it no longer overfits.
However, at some point during the training (model S with dataset SC35) it calculates NaN for the validation loss, which is then carried over for the rest of the training, as seen on the image
![Screenshot from 2021-11-16 08-46-10](https://user-images.githubusercontent.com/6875946/141943170-533b14ec-296c-42e5-8455-1955bb822efc.png)
:

The training is still successful as I'm getting 99.24% on the test set.

I suspect the following line is the culprit:
```
def validation_epoch_end(self, outs):
    avg_loss = sum([item['LOSS'] for item in outs]) / len(outs)
```
If len(outs) at any point during training evaluates to 0 then the loss becomes NaN.

**Comment by FredHaa** [comment]
The problem seems to be related to this issue:
https://github.com/pytorch/pytorch/issues/40497

I changed to model to 32-bit and have not seen NaN's since.

**Comment by ojus1** [comment]
The refactored code base no longer gets NaN, even with fp16. I think this issue is more hardware related.
